### DISCUSSION TITLE
Are we vulnerable?  (New RCE Exploit for the v3)

### DISCUSSION THREAD

**Discussion by skilau** [gtxaspec/wz_mini_hacks#659] [Q&A]
Looks like an Exploit was just released against the v3:

https://www.bleepingcomputer.com/news/security/rce-exploit-for-wyze-cam-v3-publicly-released-patch-now/

And:
https://github.com/blasty/unwyze

They don't mention our version, (we are too old I imagine).

But the cutoff seems suspiciously about the time they went from the dynamic linking of libs, to statically linking them, (and thus breaking us).

**Comment by Banewraith** [comment]
I'm curious about this.  I'm about to deploy WZ_Mini to several V3s at my parents house but will not if this exploit is an active issue.

**Comment by dudududodododedede** [comment]
fyi the vulnerability requires local LAN access i.e. the attacker must already be on your network, unless for some reason you DMZ'd the wyzecam or port-forwarded the iCamera UDP port 32761

**Comment by gtxaspec** [comment]
exploits are good, in our case typically! 

Anyway,  the RCE discussed here is limited to your LAN.  And you could always enable iptables firewall on the device with wz_mini and restrict all traffic to the camera.  I don't see it as an issue personally.

**Comment by sideup66** [comment]
If anyone wants to take a paranoid route to secure their camera, in my setup I enabled the firewall in the conf file (though it looks to be running regardless of the setting) and configured with the following script in my /opt/wz_mini/etc/rc.local.d:

#!/bin/sh
#set some basic firewall rules
/opt/wz_mini/bin/iptables -A INPUT -m state --state RELATED,ESTABLISHED -j ACCEPT
/opt/wz_mini/bin/iptables -A INPUT -i lo -j ACCEPT
/opt/wz_mini/bin/iptables -A INPUT -p tcp --dport 22 -j ACCEPT
/opt/wz_mini/bin/iptables -A INPUT -j DROP


What does this do:
1. set a rule for the INPUT table for ALL related/established connections. This is so we dont break anything the device specifically asked for outbound, also a shortcut for traffic that is related to something that was checked further down the list. Less iptables processing.
2. accept ALL traffic on loopback (this lets local services like dnsmasq through
3. allow traffic on port 22 (ssh)
4. drop everything else not related to the above rules.

This could be used if you want a basic layout to secure your device a bit and are concerned of a bad actor. Though if someone is trying this, your network is most likely compromised in larger ways. Note, using this youll have to add rules for other ports if you use services on them! Either way, I thought this might be a useful jumping off point for anyone looking to secure the device a bit.

**Comment by endertable** [comment]
This may be corny, but does anyone know the code to write to actually "run arbitray code" on the affected cams.  The reason I ask is the V2 doorbell initial FW that it ships with may be subject to this hack.  If we get telnet running, remount /etc with a new shadow db, maybe we can get in and get wz_mini on to it.  I imagine the kernel-on-the-SD-card won't work, but can maybe just flash the cam with the wz-mini kernel. There may be other implications but need to try everything if we want to succeed with having our home brews.
